internal/decryption/pem/pem.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pem
+
+import (
+	"encoding/pem"
+	"errors"
+
+	"github.com/spf13/afero"
+	"go.step.sm/crypto/pemutil"
+)
+
+type BlockType string
+
+const (
+	CertificateBlock         BlockType = "CERTIFICATE"
+	RSAPrivateKeyBlock       BlockType = "RSA PRIVATE KEY"
+	EncryptedPrivateKeyBlock BlockType = "ENCRYPTED PRIVATE KEY"
+)
+
+var (
+	ErrKMIPCertificateBlock       = errors.New("file does not contain a certificate block")
+	ErrKMIPMissingPrivateKeyBlock = errors.New("file does not contain a private key block")
+)
+
+func load(fs afero.Fs, filename string) (map[BlockType]*pem.Block, error) {
```

Comment:
passing around `fs afero.Fs` is not a great interface for users of this lib, could we have a private "default" struct that uses file system and expose afero only as part of testing, you can see something similar with how we handle profile config, you can also see something similar in the pemutil lib how they have a [private context](https://github.com/smallstep/crypto/blob/a7692cc83628e44cb7bd8621b82870c2565b93c6/pemutil/pem.go#L48-L66) to handle the common things

Reply:
made the changes (it simplified the code too)